### PR TITLE
Bump to version 0.23.0

### PR DESCRIPTION
**Pull request by tzheleznyak**
### Desired Outcome

Bump version to 0.23.0

### CODE REVIEW

File: CHANGELOG.md
Comment (diverdane) [inline CHANGELOG.md:12]: Would you mind adding an `Added` entry here that probably should have been added earlier:
```
- Add support for configuration via Pod Annotations. [[cyberark/conjur-authn-k8s-client#407](https://github.com/cyberark/conjur-authn-k8s-client/pull/407)
```

Also, would you mind changing the first entry in the `Changed` list for this version to say that we're going from Golang `v1.15` to `v1.17` (instead of `v1.16`), and add a link to a second PR. That is, change this line:
```
- The project Golang version is updated from the end-of-life v1.15 to version v1.16. [cyberark/conjur-authn-k8s-client#416](https://github.com/cyberark/conjur-authn-k8s-client/pull/416)
```
To this:
```
- The project Golang version is updated from the end-of-life v1.15 to version v1.17.
  [cyberark/conjur-authn-k8s-client#416](https://github.com/cyberark/conjur-authn-k8s-client/pull/416)
  [cyberark/conjur-authn-k8s-client#424](https://github.com/cyberark/conjur-authn-k8s-client/pull/424)
```

**Review by diverdane — COMMENTED** [review]
Looking pretty good, thanks for bumping this for us. Just a couple of nits.

**Comment by szh** [comment]
There are now breaking changes in this repo relating to the config refactoring (#425). Do we need to bump the major version?

File: bin/test-workflow/7_app_deploy.sh
Comment (tzheleznyak) [inline bin/test-workflow/7_app_deploy.sh:86]: I dont see the logs in the build maybe they should be added in a different place. Maybe after line 90

Comment (tzheleznyak) [reply bin/test-workflow/7_app_deploy.sh:86]: Should be a -c flag befpre secret;ees

File: CHANGELOG.md
Comment (diverdane) [inline CHANGELOG.md:9]: Date needs to be bumped.

**Review by diverdane — DISMISSED** [review]
Release date needs to be updated, but otherwise LGTM!!!
I clicked on Approve, hope this sticks after the date updating.

**Review by diverdane — APPROVED** [review]
LGTM!!! Onward and upward!

**Pull request merged**
